src/components/views/CheckoutPage.ts

```diff
@@ -0,0 +1,366 @@
+        paymentInputCard.setAttribute('pattern', INPUT_TYPE_CARD_NUMBER_PATTERN);
+        paymentInputCard.addEventListener('input', () => {
+            this.checkCardNumber(paymentInputCard.value);
+            paymentInputCard.value = paymentInputCard.value.slice(0, 16);
+            const paymentSstemLogoClass = this.checkCardNumber(paymentInputCard.value);
+            if (paymentSstemLogoClass) {
+                paymentSystemLogo.classList.remove('visa-logo');
+                paymentSystemLogo.classList.remove('mastercard-logo');
+                paymentSystemLogo.classList.remove('a-express-logo');
+                paymentSystemLogo.classList.remove('nocard-logo');
+                paymentSystemLogo.classList.add(paymentSstemLogoClass);
+            }
+        });
+        paymentInputCardContainer.append(paymentSystemLogo, paymentInputCard);
+
+        const paymentInputValid = createElem(HTMLTags.INPUT, 'payment-input-valid payment-input') as HTMLInputElement;
+        paymentInputValid.setAttribute('placeholder', ORDER_PLACEHOLDER_CARD_VALID);
+        paymentInputValid.setAttribute('type', INPUT_TYPE_TEXT);
+        paymentInputValid.setAttribute('required', '');
+        paymentInputValid.setAttribute('pattern', INPUT_TYPE_CARD_VALID_TIME_PATTERN);
+        paymentInputValid.addEventListener('input', () => {
+            this.checkCardValid(paymentInputValid);
+        });
+
+        const paymentInputCvv = createElem(HTMLTags.INPUT, 'payment-input-cvv payment-input') as HTMLInputElement;
+        paymentInputCvv.setAttribute('placeholder', ORDER_PLACEHOLDER_CARD_CVV);
+        paymentInputCvv.setAttribute('type', INPUT_TYPE_NUMBER);
+        paymentInputCvv.setAttribute('min', '100');
+        paymentInputCvv.setAttribute('max', '999');
+        paymentInputCvv.setAttribute('required', '');
+        paymentInputCvv.setAttribute('pattern', INPUT_TYPE_CARD_CVV_PATTERN);
+        paymentInputCvv.addEventListener('input', () => {
+            paymentInputCvv.value = paymentInputCvv.value.slice(0, 3);
+            this.checkCardCVV(paymentInputCvv.value);
+        });
+
+        paymentlInfoContent.append(paymentInputCardContainer, paymentInputValid, paymentInputCvv);
+
+        paymentContainer.append(paymentTitle, paymentlInfoContent);
+        return paymentContainer;
+    }
+
+    private destroyAllChildNodes(parent: Node) {
+        while (parent.firstChild) {
+            parent.removeChild(parent.firstChild);
+        }
+    }
+
+    public showModal() {
+        this.container.classList.add('popup_active');
+        document.body.classList.add('inactive-order');
+        document.body.addEventListener('click', (e: MouseEvent) => {
+            this.checkClickWhenOrderOpen(e);
+        });
+    }
+
+    public hideModal() {
+        this.container.classList.remove('popup_active');
+        document.body.classList.remove('inactive-order');
+        document.body.removeEventListener('click', (e: MouseEvent) => {
```

Comment:
This removeEventListener will never works. It is imposible to remove handler with arrow function.